### PR TITLE
Add Cryptocompare EAv3

### PR DESCRIPTION
**Pull request by boxhock**
## Steps to Test

1. yarn test packages/sources/cryptocompare-test/test/integration/

## Quality Assurance

- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `<ADAPTER_PACKAGE>/schemas/env.json` and `<ADAPTER_PACKAGE>/README.md`
- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [ ] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Clubhouse/Shortcut
- [ ] This is related to a maximum of one Clubhouse/Shortcut story or GitHub issue
- [ ] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)
- [ ] All code changes have 100% unit and integration test coverage. If testing is not applicable or too difficult to justify doing, the reasoning should be documented explicitly in the PR.

### CODE REVIEW

File: packages/sources/cryptocompare-test/package.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  "scripts": {
+    "clean": "rm -rf dist && rm -f tsconfig.tsbuildinfo",
+    "prepack": "yarn build",
+    "build": "tsc -b",
+    "server": "node -e 'require(\"./index.js\").server()'",
+    "server:dist": "node -e 'require(\"./dist/index.js\").server()'",
+    "start": "yarn server:dist"
+  },
+  "dependencies": {
+    "@chainlink/external-adapter-framework": "0.3.6",
```
Comment (alejoberardino) [inline packages/sources/cryptocompare-test/package.json:31]: This was merged after work on this was done, but would you mind upgrading to `0.5.1`? That version has the noticeable refactor of the type generics, and would be good to do it before merging imo

File: packages/sources/cryptocompare-test/src/endpoint/crypto-ws.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  MESSAGE: string
+  PARAMETER: string
+  INFO: string
+}
+
+const logger = makeLogger('CryptoCompareCryptoEndpoint')
+
+export const transport = new WebSocketTransport({
+  url: (context: AdapterContext) =>
+    `${DEFAULT_WS_API_ENDPOINT}?api_key=${context.adapterConfig.API_KEY}`,
```
Comment (alejoberardino) [inline packages/sources/cryptocompare-test/src/endpoint/crypto-ws.ts:40]: This does not allow the ws api endpoint to be modified with the env var from the config, is this intended?

File: packages/sources/cryptocompare-test/src/endpoint/crypto-ws.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    },
+    message(message: Message): ProviderResult<PriceEndpointParams>[] | undefined {
+      logger.trace(message, 'Got response from websocket')
+      if (message.TYPE === '5' && 'PRICE' in message) {
+        return [
+          { params: { base: message.FROMSYMBOL, quote: message.TOSYMBOL }, value: message.PRICE },
+        ]
+      }
+      if (message.MESSAGE === 'INVALID_SUB') {
+        logger.error(message, 'asset not supported by data provider')
```
Comment (alejoberardino) [inline packages/sources/cryptocompare-test/src/endpoint/crypto-ws.ts:64]: We should add a TODO here to return an error response once I add support for it

File: packages/sources/cryptocompare-test/src/endpoint/index.ts [hunk 4]
```diff
@@ -0,0 +1,2 @@
+export { endpoint as cryptoWsEndpoint } from './crypto-ws'
+export { endpoint as cryptoEndpoint } from './crypto'
```
Comment (alejoberardino) [inline packages/sources/cryptocompare-test/src/endpoint/index.ts:2]: Imma be annoying here too, to see if we can kill two birds with one stone: let's add a RoutingTransport that takes both of these, and turns it into one endpoint that chooses where to send the request based on the endpoint name ('crypto' would go to the batch one, 'crypto-ws' to the websocket one). This way we can start testing this new functionality with NOPs 😄

**Review by alejoberardino — CHANGES_REQUESTED** [review]
This looks great, left a few requests to use things from the new framework release. We can potentially do them in a separate PR, but I think it'd be easier to just change this one. Lmk

File: packages/sources/cryptocompare-test/src/endpoint/crypto-ws.ts (hunk 2 above)
Comment (boxhock) [reply packages/sources/cryptocompare-test/src/endpoint/crypto-ws.ts:40]: This is how it was done in the v3 repo, so I didn't intend anything further than that. Is there a more ideal way to do this?

Comment (boxhock) [reply packages/sources/cryptocompare-test/src/endpoint/crypto-ws.ts]: This has been updated now. Please confirm if resolved

**Review by alejoberardino — APPROVED** [review]

File: packages/sources/cryptocompare-test/src/endpoint/crypto-ws.ts (hunk 2 above)
Comment (alejoberardino) [reply packages/sources/cryptocompare-test/src/endpoint/crypto-ws.ts]: It is, ty

**Pull request merged**